app/controllers/courses_controller.rb

```diff
@@ -152,6 +152,22 @@ def save_unlocks
     redirect_to manage_unlocks_organization_course_path(@organization, @course), alert: e.to_s
   end
 
+  def student_emails
```

Comment:
This functionality seems to overlap with `EmailsController` and `ParticipantsController`. I think we should avoid duplicating the functionality and refactor the former to be suitable for what you need instead.